independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/app/CuratedApplication.java

```diff
@@ -158,7 +157,8 @@ private synchronized void processCpElement(AppArtifact artifact, Consumer<ClassP
 
     private void addCpElement(QuarkusClassLoader.Builder builder, AppArtifact dep, ClassPathElement element) {
         final AppArtifactKey key = dep.getKey();
-        if (appModel.getParentFirstArtifacts().contains(key)) {
+        if (appModel.getParentFirstArtifacts().contains(key)
```

Comment:
AppArtifact.getKey() returns a key that includes groupId:artifactId:classifier:type, while the classloading config expects groupId:artifactId. The same for reloadableArtifacts below.

Reply:
That is fine though, as it uses the String[] constructor for AppArtifactKey that will fill in the missing info if only GA is provided.